app/src/chainlit.py

```diff
@@ -118,7 +126,7 @@ async def on_message(message: cl.Message) -> None:
 
     engine: chat_engine.ChatEngineInterface = cl.user_session.get("chat_engine")
     try:
-        result = engine.on_message(question=message.content)
+        result = await cl.make_async(lambda: engine.on_message(question=message.content))()
```

Comment:
Based on https://github.com/Chainlit/chainlit/issues/274#issuecomment-1677920286, this needs to be asynchronous to avoid the frontend `Could not reach the server` alert, which pops up when the frontend times out. The backend continues to work and produces a response, but the response never reaches the UI since the frontend times out.